easybuild/easyblocks/m/metagenome_atlas_bundle.py

```diff
@@ -0,0 +1,330 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing Metagenome-Atlas, implemented as an easyblock.
+
+@author: Pavel Grochal (INUITS)
+"""
+import os
+import stat
+
+from easybuild.tools.filetools import adjust_permissions, write_file
+from easybuild.easyblocks.generic.pythonbundle import PythonBundle
+
+
+class EB_Metagenome_minus_Atlas_Bundle(PythonBundle):
+    """
+    Support for building/installing Metagenome-Atlas.
+
+    This easyblock has to be named differently from 'EB_Metagenome_minus_Atlas' due to being Bundle with extensions.
+    Otherwise it tries to install itself in extension step, which fails.
+    """
+
+    def post_install_step(self):
+        """Create snakemake config files"""
+
+        # Files were obtained by:
+        # cookiecutter https://github.com/metagenome-atlas/clusterprofile.git -o ~/.config/snakemake
+
+        snakemake_conf_dir = os.path.join(self.installdir, 'snakemake-config')
+
+        cluster_config_file = os.path.join(snakemake_conf_dir, 'cluster_config.yaml')
+        config_file = os.path.join(snakemake_conf_dir, 'config.yaml')
+        key_mapping = os.path.join(snakemake_conf_dir, 'key_mapping.yaml')
+        lsf_status = os.path.join(snakemake_conf_dir, 'lsf_status.py')
+        pbs_status = os.path.join(snakemake_conf_dir, 'pbs_status.py')
+        scheduler = os.path.join(snakemake_conf_dir, 'scheduler.py')
+        slurm_status = os.path.join(snakemake_conf_dir, 'slurm_status.py')
+
+        write_file(cluster_config_file, CLUSTER_CONFIG_FILE_TEXT)
+        write_file(config_file, CONFIG_FILE_TEXT % cluster_config_file)
+        write_file(key_mapping, KEY_MAPPING_TEXT)
+        write_file(lsf_status, LSF_STATUS_TEXT)
+        write_file(pbs_status, PBS_STATUS_TEXT)
+        # scheduler file includes small tweak from original file in git
+        write_file(scheduler, SCHEDULER_TEXT)
+        write_file(slurm_status, SLURM_STATUS_TEXT)
+
+        # make sure .py scripts are executable
+        adjust_permissions(lsf_status, stat.S_IXUSR)
+        adjust_permissions(pbs_status, stat.S_IXUSR)
+        adjust_permissions(scheduler, stat.S_IXUSR)
+        adjust_permissions(slurm_status, stat.S_IXUSR)
+
+
+CLUSTER_CONFIG_FILE_TEXT = r"""
+__default__:
+  nodes: 1
+"""
+
+CONFIG_FILE_TEXT = r"""
+restart-times: 0
+cluster-config: "%s" #abs path
+cluster: "scheduler.py" #
+cluster-status: "slurm_status.py" #
+max-jobs-per-second: 10
+max-status-checks-per-second: 10
+cores: 99 # how many jobs you want to submit to your cluster queue
+local-cores: 1
+rerun-incomplete: true  # recomended for cluster submissions
+keep-going: false
+"""
+
+KEY_MAPPING_TEXT = r"""
+# only parameters defined in key_mapping (see below) are passed to the command in the order specified.
+system: "slurm" #check if system is defined below
+
+slurm:
+  command: "sbatch --parsable"
```

Comment:
Where does this come from, did you have to compose this yourself?

Reply:
This is also taken straight from the template.
https://github.com/metagenome-atlas/clusterprofile/blob/master/{{cookiecutter.profile_name}}/key_mapping.yaml